### ISSUE TITLE
`shopt -s lastpipe` should disable SC2030/SC2031 appropriately

### ISSUE THREAD

**Issue by CyberShadow** [koalaman/shellcheck#732]
ShellCheck correctly warns on this snippet (from the SC2031 wiki):

``` bash
n=0
printf "%s\n" {1..10} | while read i; do (( n+=i )); done
echo $n
```

However, the code will behave as expected if `shopt -s lastpipe` is used, making the warning spurious.

Perhaps ShellCheck should detect this option is in use and stop treating the last command in a pipeline as a subshell, in the same way that `set -e` disables `cd` warnings.
